torchinductor/ir.py

```diff
@@ -2653,27 +2656,43 @@ def realize(self):
         self.data.name = V.graph.register_buffer(self.data)
         return self.data.name
 
+    def realize_hint(self):
+        """
+        Called on buffers we expect to be forced to realize later.
+        """
+        if self.num_reads() > 1:
+            self.realize()
+
     def mark_reuse(self, users):
         if users <= 1:
             return
         if isinstance(self.data, (Pointwise, Reduction)):
-            read_writes = ComputedBuffer(
-                name=None,
-                layout=FlexibleLayout(
-                    device=self.data.get_device(),
-                    dtype=self.data.get_dtype(),
-                    size=self.data.get_size(),
-                ),
-                data=self.data,
-            ).get_read_writes()
+            num_reads = self.num_reads()
 
             # TODO(jansel): this heuristic is a wild guess
             if (
-                len(read_writes.reads) > config.realize_reads_threshold
+                num_reads > config.realize_reads_threshold
                 or len(self.inner_fn_str()) > config.realize_bytes_threshold
             ):
                 self.realize()
 
+    def num_reads(self):
+        if isinstance(
+            self.data, (ComputedBuffer, InputsKernel, InputBuffer, ReinterpretView)
+        ):
+            return 1
```

Comment:
Why return 1 for `ComputedBuffer`? Should we recursively call `num_reads` on it?

Reply:
Fixed